src/Service.Tests/Configuration/TelemetryTests.cs

```diff
@@ -0,0 +1,295 @@
+
+        Assert.IsTrue(telemetryItems.Any(item =>
+            item is RequestTelemetry
+            && ((RequestTelemetry)item).Name.Equals("POST /graphql")
+            && ((RequestTelemetry)item).ResponseCode.Equals("200")
+            && ((RequestTelemetry)item).Url.PathAndQuery.Equals("/graphql")));
+
+        Assert.IsTrue(telemetryItems.Any(item =>
+        item is RequestTelemetry
+        && ((RequestTelemetry)item).Name.Equals("POST Rest/Insert [route]")
+            && ((RequestTelemetry)item).ResponseCode.Equals("403")
+            && ((RequestTelemetry)item).Url.PathAndQuery.Equals("/api/Publisher/id/1?name=Test")));
+
+        Assert.IsTrue(telemetryItems.Any(item =>
+        item is ExceptionTelemetry
+        && ((ExceptionTelemetry)item).Message.Equals("Authorization Failure: Access Not Allowed.")));
+    }
+
+    /// <summary>
+    /// Testing the Hosted Scenario for both configuration endpoint.
+    /// Tests that different telemetry items such as Traces or logs, Exceptions and Requests
+    /// are correctly sent to application Insights when enabled.
+    /// Also asserting on their respective properties.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow(CONFIGURATION_ENDPOINT)]
+    [DataRow(CONFIGURATION_ENDPOINT_V2)]
+    public async Task TestTelemetryItemsAreSentCorrectly_HostedScenario(string configurationEndpoint)
+    {
+        // Disable parallel execution for this test method
+        TestContext.Properties.Add("ParallelScope", "Individual");
+
+        SetUpTelemetryInconfig(CONFIG_WITH_TELEMETRY, isTelemetryEnabled: true, TEST_APP_INSIGHTS_CONN_STRING);
+
+        string[] args = new[]
+        {
+            $"--ConfigFileName={CONFIG_WITH_TELEMETRY}"
+        };
+
+        // Instantiate new server with no runtime config for post-startup configuration hydration tests.
+        List<ITelemetry> telemetryItems = new();
+        Startup.CustomTelemetryChannel = new CustomTelemetryChannel(telemetryItems);
+        TestServer server = new(Program.CreateWebHostFromInMemoryUpdateableConfBuilder(Array.Empty<string>()));
+
+        using (HttpClient client = server.CreateClient())
+        {
+            JsonContent content = GetPostStartupConfigParams(TestCategory.MSSQL, _configuration, configurationEndpoint);
+
+            HttpResponseMessage postResult =
+            await client.PostAsync(configurationEndpoint, content);
+            Assert.AreEqual(HttpStatusCode.OK, postResult.StatusCode);
+
+            HttpStatusCode restResponseCode = await GetRestResponsePostConfigHydration(client, "Publisher/id/1");
+
+            Assert.AreEqual(expected: HttpStatusCode.Forbidden, actual: restResponseCode);
+
+            HttpStatusCode graphqlResponseCode = await GetGraphQLResponsePostConfigHydration(client);
+
+            Assert.AreEqual(expected: HttpStatusCode.OK, actual: graphqlResponseCode);
+        }
+
+        server.Dispose();
```

Comment:
nit: you can use a `using` statement with TestServer setup, this is done in other tests in ConfigurationTests.cs, that way you don't need to manually dispose the server object.
```csharp
        using (TestServer server = new(Program.CreateWebHostFromInMemoryUpdateableConfBuilder(Array.Empty<string>())))
        using (HttpClient client = server.CreateClient())
        {
            ...
        }
```